### ISSUE TITLE
Cannot import @sentry/electron/main as library for main process

### ISSUE THREAD

**Issue by mx51damon** [getsentry/sentry-electron#474]
## Versions + Platform

- [3.0.7] SDK version - `@sentry/electron@v3.0.7`
- [17.2.0] Electron version - `electron@v17.2.0`
- [MacOS] Platform - `Windows/macOS/Linux`

## Description

Dear Sentry team, 

Currently, when I try to use latest version 3.0.7 and use it for my project, I faced this error below:

<img width="247" alt="image" src="https://user-images.githubusercontent.com/88181723/165204269-b8937117-a597-48cc-802a-a37adb0d890c.png">

And here is my implementation:

```js
import { processTypes } from '../constants';
import isDevEnvironment from './applicationEnvironment';

// eslint-disable-next-line import/no-mutable-exports
let Sentry;

if (process.type === processTypes.renderer) {
  Sentry = require('@sentry/electron/renderer');
} else {
  Sentry = require('@sentry/electron/main');
}

if (!isDevEnvironment) {
  Sentry.init({
    debug: isDevEnvironment,
    dsn: 'DSN_LINK',
    onFatalError: error => {
      // Global uncaught errors will be sent to sentry and then the application will exit
      console.log('Fatal', error); // eslint-disable-line no-console
      process.exit(1);
    },
  });
}

export { Sentry as default };
```

May I ask for some help with it?

Many thanks ~

**Comment by timfish** [comment]
I guess you're using a bundler and it's sees the following code and includes both the main and renderer code in the main process bundle:
```ts
if (process.type === processTypes.renderer) {
  Sentry = require('@sentry/electron/renderer');
} else {
  Sentry = require('@sentry/electron/main');
}
```

Take a look [at the examples](https://github.com/getsentry/sentry-electron/tree/master/examples) and configure the SDK independently in the main and renderer processes.

**Comment by mx51damon** [comment]
Hi @timfish, I have created 2 separated files in order to configure the main process and renderer process sentry configurations.

Here is the folder structure and code below,

![image](https://user-images.githubusercontent.com/88181723/165407049-31748e03-079b-4e74-b92b-4990f57b4310.png)


index.js:

```js
import isDevEnvironment from '../applicationEnvironment';
import { processTypes } from '../../constants';

const dsn = 'DSN_KEY_STRING';
const { sentry, initialiseSentry } = process.type === processTypes.renderer ? require('./renderer') : require('./main');

initialiseSentry(dsn, isDevEnvironment);

export { sentry as default };
```

main.js:

```js
import * as sentry from '@sentry/electron/main';
import logging from '../../main/logger';

const logger = logging.child({ service: 'spice.error' });

function logError(error, hint) {
  try {
    logger.error({ message: 'Uncaught error encountered', data: { error, hint } });
  } catch (e) {
    console.error(error); // eslint-disable-line no-console
  }
}

function initialiseSentry(dsn, isDevEnvironment) {
  sentry.init({
    debug: isDevEnvironment,
    dsn,
    beforeSend: (event, error) => {
      logError(event, error.originalException || error.syntheticException);

      // Returning null will trigger Sentry not to report the error back
      return !isDevEnvironment ? event : null;
    },
    onFatalError: async error => {
      // Global uncaught errors will be sent to sentry and then the application will exit
      logError(error);
      process.exit(1);
    },
  });
}

export { sentry, initialiseSentry };
```

renderer.js:

```js
import * as sentry from '@sentry/browser';

function initialiseSentry(dsn, isDevEnvironment) {
  sentry.init({
    debug: isDevEnvironment,
    dsn,
  });
}

export { sentry, initialiseSentry };
```

However, I still got the same issue, the error is exactly same as the first comment image shown ...

After I read the npm documentation, it seems like I can use:

Captures Node errors in the main process (using [@sentry/node](https://github.com/getsentry/sentry-javascript/tree/master/packages/node))
Captures JavaScript errors in renderer processes (using [@sentry/browser](https://github.com/getsentry/sentry-javascript/tree/master/packages/browser))

instead of using @sentry/electron/main and @sentry/electron/renderer, is this ok for me to do?

Or you have another option for me to give it a try?

Thank you for help 🙏

**Comment by timfish** [comment]
You are getting the same error because you are still selecting the main/renderer SDK entry point at runtime.

Whenever you check `process.type` this is runtime detection. Your bundler doesn't know what value to expect and subsequently includes both code paths in the bundle.

You need to remove `index.js` completely and call main/renderer from the earliest point in each process.

You could use the browser/node SDKs but you'd still have similar issues with bundler + runtime detection and you'd lose all the additional features the Electron SDK adds including:
- Native crash reporting
- Additional context and breadcrumbs
- Merging of scope and context from multiple processes

**Comment by mx51damon** [comment]
Hi @timfish, I have tried the way you mentioned above, I put the sentry configuration files respectively for main and renderer folder in my codebase, and I still get the same error. 

The most wired part is like when I switched the @sentry/electron/main to @sentry/electron/renderer, the error is disappeared. 

Also, I have checked my webpack config file, the earliest entry point for main process is  `'./src/main/index.js',`

```js
entry: [
    IS_DEV && 'webpack/hot/poll?1000',
    IS_DEV && 'source-map-support/register',
    IS_DEV && './build/electron.js',
    './src/main/index.js',
  ].filter(Boolean),
```

And also, I have used `DefinePlugin` to separate the process type whether it's a problem or not?

```js
plugins: common.plugins.concat([
    new webpack.DefinePlugin({
      'process.type': '"main"',
    }),
  ]),
```

To sum up, is there any options for me to try it on? (I hope you don't mind, I really need to get some help here)

Mega thanks @timfish.

**Comment by mx51damon** [comment]
Hi @timfish, I am not 100% sure I fixed my issue or not, but when I put sentry configuration file as part of entry configuration in webpack, the error is gone ...

```js
 entry: [
    IS_DEV && 'webpack/hot/poll?1000',
    IS_DEV && 'source-map-support/register',
    IS_DEV && './build/electron.js',
    './src/main/sentry.js',
    './src/main/index.js',
  ].filter(Boolean),
```

Something like this. Thanks for help mate 👍
